### DISCUSSION TITLE
What's the current progress of getting html, css, and json working with eglot?

### DISCUSSION THREAD

**Discussion by zeta-00** [joaotavora/eglot#749] [General]
https://github.com/joaotavora/eglot/pull/204/files
^
I really need eglot to work with these languages, are you close to merging this pull-request?

**Comment by zeta-00** [comment]
For now, I'll stick to lsp-mode for these languages in the mean time. If you merge this pull-request and the eglot works with these languages, then I'll switch them back to eglot.
Thanks in advance for the help!

**Reply by joaotavora** [reply]
@zeta-00 this PR is little more than a setting something in the customizable variable `eglot-server-programs`.  If these settings work for you and you're in such a hurry, why not do them yourself?

You see Eglot supports these language servers and any other language server that speaks LSP.  In Eglot, there are no `eglot-language-foo-languages.el` files needed.  If you want very urgently to use the features of the PR you point to, just customize the variable `eglot-server-programs` and add the things there that this PR adds.

But I'll see about merging this in the near future.  I've little time on my hands right now.

**Reply by skangas** [reply]
In case anyone else stumbles upon this, it has now been merged.

**Reply by zeta-00** [reply]
@skangas ,

Uhh, on my end Eglot is still not working with html-mode:

Emacs version: 27.2

Installed Node Packages:
nodePackages.vscode-css-languageserver-bin
nodePackages.vscode-html-languageserver-bin
nodePackages.vscode-json-languageserver
 
What html-language-server did you install, and what .emacs config did you set to get it to work?

;;------------------------------------------------------------------
(use-package eglot
  :defer t
  :hook
  (html-mode . eglot-ensure)
  (css-mode . eglot-ensure)
  (js-mode . eglot-ensure))
;;------------------------------------------------------------------

On my end the html-language-server binary is not being recognized by eglot, which is strange.

**Comment by zeta-00** [comment]
`Error in post-command-hook (#[0 "\303\304\301\242\305#\210\300\306!\205r\211q\210
?\205\307\310\311 \")\207" [#<buffer show.html> (#0) eglot--managed-mode remove-hook post-command-hook nil buffer-live-p apply eglot--connect eglot--guess-contact] 4]): (error "None of ’vscode-html-language-server, html-languageserver’ are valid executables")`
